src/index/txospenderindex.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_INDEX_TXOSPENDERINDEX_H
+#define BITCOIN_INDEX_TXOSPENDERINDEX_H
+
+#include <index/base.h>
+
+static constexpr bool DEFAULT_TXOSPENDERINDEX{false};
+
+/**
+ * TxoSpenderIndex is used to look up which transaction spent a given output.
+ * The index is written to a LevelDB database and, for each input of each transaction in a block,
+ * records the outpoint that is spent and the hash of the spending transaction.
+ */
+class TxoSpenderIndex final : public BaseIndex
+{
+protected:
+    class DB;
+
+private:
+    const std::unique_ptr<DB> m_db;
+
+    bool AllowPrune() const override { return false; }
+
+protected:
+    bool CustomAppend(const interfaces::BlockInfo& block) override;
```

Comment:
done in https://github.com/bitcoin/bitcoin/pull/24539/commits/5957f992b7afe20dca250c398bb28ad0ae52f868

I may have missed something, but it seems that if the chain tip is reorged out, `CustomRewind` won't be called until a new block is added to the chain. Until then, our txospenderindex (and I guess this might be true for other indexes) will return stale results. In our case, you'll be able to see that the spending tx has 0 confirmations and could be replaced, but it makes me wonder how usable this index is if you cannot tell if spending txs are confirmed or not. So yes technically `txospenderindex` does not require `txindex` but could it really be usable in practice without it ?